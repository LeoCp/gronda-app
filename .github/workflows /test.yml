name: Tests

on:
  push:
    branches:
      - master
jobs:
  publish:
    name: Build
    runs-on: ubuntu-latest
    steps:
      - name: 🏗  Setup repo
        uses: actions/checkout@v2

      - name: 🏗  Setup Node
        uses: actions/setup-node@v2
        with:
          node-version: 16.x
          cache: yarn

      - name: 📦  Install dependencies
        run: yarn install

      - name: 🧪  Run test
        run: yarn coveralls

      - name: Coveralls
        uses: coverallsapp/github-action@master
        with:
          github-token: ${{ secrets.GITHUB_TOKEN }}
