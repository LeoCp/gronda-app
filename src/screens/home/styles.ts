import styled from 'styled-components/native';

export const Wrapper = styled.SafeAreaView`
  flex: 1;
  background-color: #fff;
`;

export const Header = styled.View`
  padding-top: 15px;
  padding-bottom: 12px;
  align-items: center;
`;
