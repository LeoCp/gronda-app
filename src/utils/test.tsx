import React from 'react';
import { render } from '@testing-library/react-native';

import { theme } from '@utils/theme';
import { ThemeProvider } from 'styled-components/native';

export function renderWithTheme(children: React.ReactElement) {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
}
