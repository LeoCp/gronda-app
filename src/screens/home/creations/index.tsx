import React from 'react';

import { ErrorMessage } from '@screens/home/creations/components/error-message';
import { useCreations } from '@screens/home/creations/hooks/creations';

import * as S from './styles';

export function Creations() {
  const creations = useCreations();

  if (creations.isLoading) {
    return (
      <S.LoadingWrapper>
        <S.Loading />
      </S.LoadingWrapper>
    );
  }

  if (creations.error) {
    return <ErrorMessage tryAgain={creations.refetch} />;
  }

  return (
    <S.Creations
      data={creations.data || []}
      renderItem={creations.renderCreationItem}
    />
  );
}
