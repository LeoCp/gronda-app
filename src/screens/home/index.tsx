import React from 'react';
import { StatusBar } from 'react-native';

import { Icon } from '@components/icon';
import { Creations } from '@screens/home/creations';

import * as S from './styles';

export function Home() {
  return (
    <S.Wrapper>
      <StatusBar barStyle="dark-content" />
      <S.Header>
        <Icon name="logo" height={45} width={130} />
      </S.Header>
      <Creations />
    </S.Wrapper>
  );
}
