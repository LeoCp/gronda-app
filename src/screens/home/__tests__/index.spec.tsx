import React from 'react';

import { Home } from '@screens/home';
import { renderWithTheme } from '@utils/test';

jest.mock('@screens/home/creations/hooks/creations', () => ({
  useCreations: jest.fn().mockReturnValue({
    isLoading: false,
    error: false,
    refetch: jest.fn(),
    data: [],
  }),
}));

describe('Home', () => {
  it('Should match snapshot', () => {
    const { toJSON } = renderWithTheme(<Home />);
    expect(toJSON()).toMatchSnapshot();
  });
});
