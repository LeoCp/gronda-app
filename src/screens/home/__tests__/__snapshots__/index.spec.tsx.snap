// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Should match snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#fff",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "paddingBottom": 12,
        "paddingTop": 15,
      }
    }
  >
    <Image
      source={
        {
          "testUri": "../../../src/assets/icons/logo.png",
        }
      }
      style={
        {
          "height": 45,
          "width": 130,
        }
      }
    />
  </View>
  <View
    style={
      {
        "flex": 1,
        "overflow": "hidden",
      }
    }
  >
    <RCTScrollView
      ListHeaderComponent={<Banner />}
      applyWindowCorrection={[Function]}
      canChangeSize={true}
      contentContainerStyle={
        {
          "backgroundColor": undefined,
          "minHeight": 1,
          "minWidth": 1,
          "paddingLeft": 8,
          "paddingRight": 8,
        }
      }
      contentHeight={0}
      contentWidth={0}
      data={[]}
      dataProvider={
        DataProvider {
          "_data": [],
          "_firstIndexToProcess": 0,
          "_hasStableIds": true,
          "_requiresDataChangeHandling": false,
          "_size": 0,
          "getStableId": [Function],
          "rowHasChanged": [Function],
        }
      }
      disableRecycling={false}
      estimatedItemSize={200}
      externalScrollView={[Function]}
      finalRenderAheadOffset={250}
      forceNonDeterministicRendering={true}
      horizontal={false}
      initialOffset={0}
      initialRenderIndex={0}
      isHorizontal={false}
      keyExtractor={[Function]}
      layoutProvider={
        GridLayoutProviderWithProps {
          "_acceptableRelayoutDelta": 1,
          "_getHeightOrWidth": [Function],
          "_getLayoutTypeForIndex": [Function],
          "_getSpan": [Function],
          "_hasExpired": false,
          "_maxSpan": 2,
          "_setLayoutForType": [Function],
          "_tempDim": {
            "height": 0,
            "width": 0,
          },
          "averageWindow": AverageWindow {
            "currentAverage": 200,
            "currentCount": 1,
            "inputValues": [
              200,
            ],
            "nextIndex": 1,
          },
          "defaultEstimatedItemSize": 100,
          "layoutObject": {
            "size": undefined,
            "span": undefined,
          },
          "props": {
            "ListHeaderComponent": <Banner />,
            "contentContainerStyle": {
              "paddingHorizontal": 8,
            },
            "data": [],
            "estimatedItemSize": 200,
            "keyExtractor": [Function],
            "numColumns": 2,
            "renderItem": undefined,
            "style": {},
            "testID": "creations",
            "theme": {
              "color": {
                "black": "#000000",
                "light": "#FFFFFF",
                "primary": "#52a8c9",
                "sea-green": "#075669",
              },
              "fontFamily": {
                "bold": "TT Commons DemiBold",
                "regular": "TT Commons Regular",
              },
              "fontSize": {
                "lg": 24,
                "md": 19,
                "sm": 17,
                "xl": 60,
                "xs": 13,
              },
            },
          },
          "renderWindowInsets": {
            "height": 0,
            "width": -16,
          },
          "shouldRefreshWithAnchoring": true,
        }
      }
      maxRenderAhead={750}
      numColumns={2}
      onEndReached={[Function]}
      onEndReachedThreshold={0}
      onEndReachedThresholdRelative={0}
      onItemLayout={[Function]}
      onLayout={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onSizeChanged={[Function]}
      onVisibleIndicesChanged={[Function]}
      removeClippedSubviews={false}
      renderAheadOffset={0}
      renderAheadStep={250}
      renderContentContainer={[Function]}
      renderItemContainer={[Function]}
      rowRenderer={[Function]}
      scrollEventThrottle={16}
      scrollThrottle={16}
      scrollViewProps={
        {
          "contentContainerStyle": {
            "backgroundColor": undefined,
            "minHeight": 1,
            "minWidth": 1,
            "paddingLeft": 8,
            "paddingRight": 8,
          },
          "onLayout": [Function],
          "onScrollBeginDrag": [Function],
          "refreshControl": undefined,
          "style": {
            "minHeight": 1,
            "minWidth": 1,
          },
        }
      }
      style={
        {
          "minHeight": 1,
          "minWidth": 1,
        }
      }
      suppressBoundedSizeException={true}
      testID="creations"
      theme={
        {
          "color": {
            "black": "#000000",
            "light": "#FFFFFF",
            "primary": "#52a8c9",
            "sea-green": "#075669",
          },
          "fontFamily": {
            "bold": "TT Commons DemiBold",
            "regular": "TT Commons Regular",
          },
          "fontSize": {
            "lg": 24,
            "md": 19,
            "sm": 17,
            "xl": 60,
            "xs": 13,
          },
        }
      }
      windowCorrectionConfig={
        {
          "applyToInitialOffset": true,
          "applyToItemScroll": true,
          "value": {
            "endCorrection": 0,
            "startCorrection": 0,
            "windowShift": -1,
          },
        }
      }
    >
      <View>
        <View
          style={
            {
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              {
                "paddingLeft": 8,
                "paddingTop": 0,
              }
            }
          />
          <View
            style={
              [
                undefined,
                undefined,
              ]
            }
          >
            <View
              style={
                {
                  "paddingBottom": 8,
                  "paddingLeft": 8,
                  "paddingRight": 8,
                  "paddingTop": 8,
                }
              }
            >
              <View
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    {
                      "borderBottomLeftRadius": 8,
                      "borderBottomRightRadius": 8,
                      "borderTopLeftRadius": 8,
                      "borderTopRightRadius": 8,
                      "height": 170,
                      "marginTop": 15,
                      "width": "100%",
                    },
                  ]
                }
              >
                <FastImageView
                  defaultSource={null}
                  resizeMode="cover"
                  source={
                    {
                      "priority": "normal",
                      "uri": "https://d3566jsyo19arr.cloudfront.net/banner/marco_mueller_banner.jpg",
                    }
                  }
                  style={
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
                <View
                  style={
                    {
                      "flexBasis": 0,
                      "flexGrow": 1,
                      "flexShrink": 1,
                      "justifyContent": "center",
                      "paddingLeft": 20,
                    }
                  }
                >
                  <Text
                    bold={true}
                    color="light"
                    size="xs"
                    style={
                      [
                        {
                          "color": "#FFFFFF",
                          "fontFamily": "TT Commons DemiBold",
                          "fontSize": 13,
                        },
                        {},
                      ]
                    }
                  >
                    NEW
                  </Text>
                  <Text
                    bold={true}
                    color="light"
                    size="lg"
                    style={
                      [
                        {
                          "color": "#FFFFFF",
                          "fontFamily": "TT Commons DemiBold",
                          "fontSize": 24,
                        },
                        {
                          "marginBottom": 12,
                          "marginTop": 6,
                        },
                      ]
                    }
                  >
                    Fish preparation like
a star chef
                  </Text>
                  <Text
                    bold={false}
                    color="light"
                    size="sm"
                    style={
                      [
                        {
                          "color": "#FFFFFF",
                          "fontFamily": "TT Commons Regular",
                          "fontSize": 17,
                        },
                        {},
                      ]
                    }
                  >
                    With Rolf Fliegauf
                  </Text>
                </View>
              </View>
              <Text
                bold={true}
                color="black"
                size="md"
                style={
                  [
                    {
                      "color": "#000000",
                      "fontFamily": "TT Commons DemiBold",
                      "fontSize": 19,
                    },
                    {
                      "marginBottom": 16,
                      "marginTop": 30,
                    },
                  ]
                }
              >
                Creation for you
              </Text>
            </View>
          </View>
          <AutoLayoutView
            enableInstrumentation={false}
            horizontal={false}
            onBlankAreaEvent={[Function]}
            onLayout={[Function]}
            renderAheadOffset={0}
            scrollOffset={0}
            style={
              {
                "height": 0,
                "width": 0,
              }
            }
            windowSize={0}
          />
          <CellContainer
            index={-1}
            style={
              [
                undefined,
                undefined,
              ]
            }
          />
          <View
            style={
              {
                "paddingBottom": 0,
                "paddingRight": 8,
              }
            }
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
